trieste/acquisition/rule.py

```diff
@@ -1097,6 +1105,225 @@ def state_func(
+
+        self._local_models: Mapping[Tag, TrainableSupportsGetKernel]
+        if self._local_models.keys() != {OBJECTIVE}:
+            raise ValueError(
+                f"dict of models must contain the single key {OBJECTIVE}, got keys {models.keys()}"
+            )
+
+        if datasets is None or datasets.keys() != {OBJECTIVE}:
+            raise ValueError(
+                f"""datasets must be provided and contain the single key {OBJECTIVE}"""
+            )
+
+        dataset = datasets[OBJECTIVE]
+        local_model = self._local_models[OBJECTIVE]
+
+        global_lower = search_space.lower
+        global_upper = search_space.upper
+
+        y_min = tf.reduce_min(dataset.observations, axis=0)
+
+        def state_func(
+            state: TURBO.State | None,
+        ) -> tuple[TURBO.State | None, TensorType]:
+            if state is None:  # initialise first TR
+                L, failure_counter, success_counter = self._L_init, 0, 0
+            else:  # update TR
+                step_is_success = y_min < state.y_min - 1e-10  # maybe make this stronger?
+                failure_counter = (
+                    0 if step_is_success else state.failure_counter + 1
+                )  # update or reset counter
+                success_counter = (
+                    state.success_counter + 1 if step_is_success else 0
+                )  # update or reset counter
+                L = state.L
+                if success_counter == self._success_tolerance:
+                    L *= 2.0  # make region bigger
+                    success_counter = 0
+                elif failure_counter == self._failure_tolerance:
+                    L *= 0.5  # make region smaller
+                    failure_counter = 0
+
+                L = tf.minimum(L, self._L_max)
+                if L < self._L_min:  # if gets too small then start again
+                    L, failure_counter, success_counter = self._L_init, 0, 0
+
+            # build region with volume according to length L but stretched according to lengthscales
+            xmin = dataset.query_points[tf.argmin(dataset.observations)[0], :]  # centre of region
+            lengthscales = (
+                local_model.get_kernel().lengthscales
+            )  # stretch region according to model lengthscales
+            tr_width = (
+                lengthscales * L / tf.reduce_prod(lengthscales) ** (1.0 / global_lower.shape[-1])
+            )  # keep volume fixed
+            acquisition_space = Box(
+                tf.reduce_max([global_lower, xmin - tr_width / 2.0], axis=0),
+                tf.reduce_min([global_upper, xmin + tr_width / 2.0], axis=0),
+            )
+
+            # fit the local model using just data from the trust region
+            local_dataset = get_local_dataset(acquisition_space, dataset)
+            local_model.update(local_dataset)
+            local_model.optimize(local_dataset)
+
```

Comment:
here you're possibly missing

`self._local_models[OBJECTIVE] = local_model`

Reply:
@uri-granta experimenting with TURBO I found that  if I use this local model to plot some quantities during the BO run e.g.

`local_model = ask_tell._acquisition_rule._local_models['OBJECTIVE']
local.model.predict(...)`

the predictions do not get updated unless I add the line above. Does it make any sense to you?

Reply:
I'm not managing to reproduce the issue (the model seems to be updating fine for me). Can you push a temporary notebook or test so I can see how you're setting things up?